libminifi/test/sql-tests/mocks/MockConnectors.cpp

```diff
@@ -0,0 +1,386 @@
+  std::vector<std::string> col_names;
+  std::vector<DataType> col_types;
+  for (const auto& col_with_type : columns_with_type) {
+    auto splitted = minifi::utils::StringUtils::splitAndTrim(col_with_type, " ");
+    col_names.push_back(splitted[0]);
+    col_types.push_back(stringToDataType(splitted[1]));
+  }
+  tables_.emplace(table_name, MockRowset{col_names, col_types});
+  storeDb();
+}
+
+void MockDB::insertInto(const std::string& query, const std::vector<std::string>& args) {
+  std::string replaced_query = query;
+  for (const auto& arg : args) {
+    replaced_query = minifi::utils::StringUtils::replaceOne(replaced_query, "?", arg);
+  }
+
+  std::smatch match;
+  std::regex expr("insert into (\\w+)\\s*(\\((.*)\\))*\\s*values\\s*\\((.+)\\)");
+  std::regex_search(replaced_query, match, expr);
+  std::string table_name = match[1];
+  std::vector<std::string> values = minifi::utils::StringUtils::splitAndTrim(match[4], ",");
+  for (auto& value : values) {
+    value = minifi::utils::StringUtils::removeFramingCharacters(value, '\'');
+  }
+  auto insert_col_names = minifi::utils::StringUtils::splitAndTrim(match[3], ",");
+  if (!insert_col_names.empty()) {
+    auto col_names = tables_.at(table_name).getColumnNames();
+    std::vector<std::string> row;
+    for (const auto& col_name : col_names) {
+      auto it = std::find(insert_col_names.begin(),  insert_col_names.end(), col_name);
+      if (it != insert_col_names.end()) {
+        row.push_back(values.at(it-insert_col_names.begin()));
+      } else {
+        row.push_back("NULL");
+      }
+    }
+    tables_.at(table_name).addRow(row);
+  } else {
+    tables_.at(table_name).addRow(values);
+  }
+
+  storeDb();
+}
+
+std::unique_ptr<Rowset> MockDB::select(const std::string& query, const std::vector<std::string>& args) {
+  std::string replaced_query = query;
+  for (const auto& arg : args) {
+    replaced_query = minifi::utils::StringUtils::replaceOne(replaced_query, "?", arg);
+  }
+
+  std::smatch match;
+  std::regex expr("select\\s+(.+)\\s+from\\s+(\\w+)\\s*(where ((.+(?= order by))|.+$))*\\s*(order by (.+))*");
+  std::regex_search(replaced_query, match, expr);
+  auto cols = minifi::utils::StringUtils::splitAndTrim(match[1], ",");
+  if (cols[0] == "*") {
+    cols = {};
+  }
+  std::string table_name = match[2];
+  std::string condition_str = match[4];
+  std::function<bool(const MockRow&)> condition;
+  if (!condition_str.empty()) {
+    if (condition_str == "int_col > 103") {
+      condition = [&](const MockRow& row){ return std::atoi(row.getValue("int_col").c_str()) > 103; };
+    } else if (condition_str == "int_col > 102") {
+      condition = [&](const MockRow& row){ return std::atoi(row.getValue("int_col").c_str()) > 102; };
+    } else if (condition_str == "int_col = 11") {
+      condition = [&](const MockRow& row){ return std::atoi(row.getValue("int_col").c_str()) == 11; };
+    } else if (condition_str == "int_col = 11 and text_col = banana") {
```

Comment:
Shouldn't "banana" be in apostrophes here? Unless it's another column.

Reply:
When replacing arguments I did not add apostrophes so it worked like this. The code was removed in favor of 96c3a9dea8e0d9c0492ccf5850184755651418e9